mongo/mongo.go

```diff
@@ -27,37 +31,55 @@ const imagesLockCol = "images_locks"
 type Mongo struct {
 	Collection   string
 	Database     string
-	Session      *mgo.Session
+	Connection   *dpMongoDriver.MongoConnection
 	URI          string
+	Username     string
+	Password     string
 	client       *dpMongoHealth.Client
 	healthClient *dpMongoHealth.CheckMongoClient
 	lockClient   *dpMongoLock.Lock
+	IsSSL        bool
 }
 
 // Init creates a new mgo.Session with a strong consistency and a write mode of "majority".
-func (m *Mongo) Init(ctx context.Context) (err error) {
-	if m.Session != nil {
-		return errors.New("session already exists")
+func (m *Mongo) getConnectionConfig(shouldEnableReadConcern, shouldEnableWriteConcern bool) *dpMongoDriver.MongoConnectionConfig {
```

Comment:
The go doc is no longer correct as this function doesn't create an mgo.Session (Line above this one)